### ISSUE TITLE
SSR, ARM and Arch Linux

### ISSUE THREAD

**Issue by janhenhan** [SoundScapeRenderer/ssr#273]
Hi,

After fixing my Mac install the other week, I'd like to install SSR on a Raspberry Pi running a minimal Manjaro install, which is an Arch Linux based operating system. I saw in the SSR documentation that there even is a SSR package one can install with pacman. It installed without a complain, starts, but produces jack xruns like crazy even before actually starting to process any audio. Something is fishy there.
I then realised the package only claims support for x86 architectures, and so does the ssr-git from aur which I tried to build manually like the SSR documentation suggests. The latter at least complained explicitly about no support for aarch64.

What are my chances of making SSR run on Manjaro and ARM? If not 0, what approach would you recommend?

As always thanks a lot for your help,
Jan

**Comment by mgeier** [comment]
I'm not sure about the chances ... I think most people have used the SSR on x86 architectures so far.

I think most of the code should work independent of the processor architecture, but there are a few things to look out for:

* the convolver is optimized using SSE instructions. There is a fallback implementation, but this is hardly tested and it might be significantly slower.
* the realtime FIFO queue might not work correctly on ARM, but I'm not sure about that
* there might  be other issues that I'm not aware of

You should definitely try to compile the SSR on your own if the pre-built packages don't work for you.

And then, you should run it and report any problems you have!

If it happens to work, that's great, but if it doesn't, I don't think it is likely to get fixed unless someone steps forward and provides the necessary fixes.

**Comment by janhenhan** [comment]
Ok good news - compiling seems to have been successful. It took me a bit to figure out the right dependencies, but now I've got a ssr-brs binary that no longer shows the uncalled for xruns I saw with the SSR package from pacman. I still ran into problems (xruns) when connecting an older PD controller patch that was still using the legacy network interface.. I wanted to look into the updated network interfaces anyhow and just managed to pipe in head tracker updates via FUDI, so things are looking promising. Thanks for your input Matthias!
